include/AudioEngine.h

```diff
@@ -50,14 +50,15 @@ class AudioPort;
 class AudioEngineWorkerThread;
 
 
-const fpp_t MINIMUM_BUFFER_SIZE = 32;
-const fpp_t DEFAULT_BUFFER_SIZE = 256;
+constexpr fpp_t MINIMUM_BUFFER_SIZE = 32;
+constexpr fpp_t DEFAULT_BUFFER_SIZE = 256;
+constexpr fpp_t MAXIMUM_BUFFER_SIZE = 4096;
```

Comment:
Where is that constant from? How do you know it is 4096? I assume that it is the maximum value which `SetupDialog::m_bufferSizeSlider` can have? In this case, I would add an `assert(m_bufferSizeSlider::maximum() == 4096)` or something similar there - because if the slider's maximum changes in the future, we could risk buffer overruns in `VstEffect.cpp`?

Reply:
@sakertooth said 4096 should be used as the maximum buffer size. It's the maximum buffer size in the future if we stop processing buffer sizes greater than 256 as multiple batches of 256 frame buffers.

Reply:
> because if the slider's maximum changes in the future, we could risk buffer overruns in VstEffect.cpp?

We do get 4096 from the slider:

```cpp
m_bufferSizeSlider->setRange(1, 128);
m_bufferSizeSlider->setTickInterval(8);
m_bufferSizeSlider->setPageStep(8);
m_bufferSizeSlider->setValue(m_bufferSize / BUFFERSIZE_RESOLUTION); //BUFFERSIZE_RESOULTION = 32
...
m_bufferSize = value * BUFFERSIZE_RESOLUTION;
```

The slider should ideally be calling `setMinimum` and `setMaximum` to specify the minimum and maximum respectively to make this more clear, but the maximum does end up being 4096.

Maybe it was a mistake on my end to suggest this so early and in this PR. If I am going to change the code so that the actual buffer sizes are used, maybe its best I handle this all in my own PR since it seems like this may result in out of scope changes.

I feel very bad now this @messmerd went through all of that trouble to do what I mentioned, but it's probably better I take note of the previous code and make sure I change it in my branch. I'll probably notice that the code would have to be changed anyways since the PR that I'm doing concerns a good chunk of the audio processing code.

Reply:
I don't think there was a mistake. Why not just add my suggested `assert`? I would be perfectly fine with that.

Reply:
I just made a one line change in SetupDialog.cpp to use the new `MAXIMUM_BUFFER_SIZE` constant in the buffer size slider's range. We should be good now